### PR TITLE
support to new design the API golang.org/x/oauth2

### PR DESCRIPTION
**Pull request by weslleyandrade**
the project "golang/oauth2" recommended this change.

### CODE REVIEW

File: oauth2.go [hunk 1]
```diff
@@ -166,14 +170,14 @@ var LoginRequired martini.Handler = func() martini.Handler {
...
-func login(f *oauth2.Options, s sessions.Session, w http.ResponseWriter, r *http.Request) {
+func login(f *oauth2.Config, s sessions.Session, w http.ResponseWriter, r *http.Request) {
 	next := extractPath(r.URL.Query().Get(keyNextPage))
 	if s.Get(keyToken) == nil {
 		// User is not logged in.
 		if next == "" {
 			next = "/"
 		}
-		http.Redirect(w, r, f.AuthCodeURL(next, "", ""), codeRedirect)
+		http.Redirect(w, r, f.AuthCodeURL(next, AccessType), codeRedirect)
```
Comment (rakyll) [inline oauth2.go]: http.Redirect(w, r, f.AuthCodeURL(next), codeRedirect)

File: oauth2.go [hunk 2]
```diff
@@ -186,18 +190,17 @@ func logout(s sessions.Session, w http.ResponseWriter, r *http.Request) {
...
+func handleOAuth2Callback(f *oauth2.Config, s sessions.Session, w http.ResponseWriter, r *http.Request) {
 	next := extractPath(r.URL.Query().Get("state"))
 	code := r.URL.Query().Get("code")
-	t, err := f.NewTransportFromCode(code)
+	t, err := f.Exchange(oauth2.NoContext, code)
 	if err != nil {
 		// Pass the error message, or allow dev to provide its own
 		// error handler.
 		http.Redirect(w, r, PathError, codeRedirect)
-		return
```
Comment (rakyll) [inline oauth2.go]: return?

Comment (weslleyandrade) [inline oauth2.go]: was removed by mistake, sorry.
in the commit 939de0b the "return" was fixed

**Comment by weslleyandrade** [comment]
@rakyll Again I corrected the requested notes. I took the opportunity to incorporate the changes made to the API (https://go-review.googlesource.com/#/c/2195/)

**Comment by shackra** [comment]
Uh, hope this get merged soon! my issue #47 is really a stopper

File: oauth2.go (hunk 2 above)
Comment (rakyll) [inline oauth2.go]: The diff says it's still removed.

**Comment by rakyll** [comment]
Please squash your commits.

**Comment by weslleyandrade** [comment]
done.

File: oauth2.go (hunk 2 above)
Comment (rakyll) [inline oauth2.go]: Revert this line please.

Comment (weslleyandrade) [inline oauth2.go]: sorry you were right.
in the commit 0ae7a12bf7dff68a3d1682ca88d58340577b1d6f  was fixed

**Comment by dougbarrett** [comment]
Can we get this merged? I'm unable to deploy correctly on heroku because of this...

**Comment by weslleyandrade** [comment]
@dougbarrett you're right
in my last commiti against the correction

@rakyll you think necessary squash

**Comment by dougbarrett** [comment]
@weslleyandrade @rakyll just wondering what the status of this is? The live package is broken at this point, and this will fix it.

Thanks!

File: oauth2.go [hunk 3]
```diff
@@ -91,52 +87,57 @@ func (t *token) String() string {
 }
 
 // Google returns a new Google OAuth 2.0 backend endpoint.
-func Google(opt ...oauth2.Option) martini.Handler {
-	return NewOAuth2Provider(append(opt, oauth2.Endpoint(
-		"https://accounts.google.com/o/oauth2/auth",
-		"https://accounts.google.com/o/oauth2/token"),
-	))
+func Google(conf *oauth2.Config) martini.Handler {
+	conf.Endpoint = oauth2.Endpoint{
```
Comment (rakyll) [inline oauth2.go]: Use google.Endpoint. http://godoc.org/golang.org/x/oauth2/google#pkg-variables

File: oauth2.go [hunk 4]
```diff
@@ -91,52 +87,57 @@ func (t *token) String() string {
...
-	return NewOAuth2Provider(append(opt, oauth2.Endpoint(
-		"https://github.com/login/oauth/authorize",
-		"https://github.com/login/oauth/access_token"),
-	))
+func Github(conf *oauth2.Config) martini.Handler {
+	conf.Endpoint = oauth2.Endpoint{
+		AuthURL:  "https://github.com/login/oauth/authorize",
+		TokenURL: "https://github.com/login/oauth/access_token",
+	}
+
```
Comment (rakyll) [inline oauth2.go]: These arbitrary blank lines are distracting. I'd remove them all around.

**Comment by rakyll** [comment]
Please squash and address the comments I've left, then we can merge.

**Comment by weslleyandrade** [comment]
@rakyll Done.

**Comment by rakyll** [comment]
Thanks.

**Pull request merged**